src/Storages/StorageS3.cpp

```diff
@@ -1035,7 +1043,9 @@ Pipe StorageS3::read(
 
     auto read_from_format_info = prepareReadingFromFormat(column_names, storage_snapshot, supportsSubsetOfColumns(), getVirtuals());
 
-    const size_t max_download_threads = local_context->getSettingsRef().max_download_threads;
+    const size_t max_parsing_threads = local_context->getSettingsRef().max_threads;
```

Comment:
Lets name that tmp var just `max_threads`.
Leave the name `max_parsing_threads` for StorageS3Source class in that file. Otherwise it is confusing. That value is not the same.

The next var `parsing_threads` could be named as `max_parsing_threads` without confusing. Because that var is passed to the StorageS3Source class as `max_parsing_threads`. But that is also OK, as it is.